enginetest/queries/query_plans.go

```diff
@@ -5539,36 +5597,32 @@ CROSS JOIN
 			" │           │   │       └─ Table(HGMQ6)\n" +
 			" │           │   └─ TableAlias(sn)\n" +
 			" │           │       └─ Table(NOXN3)\n" +
-			" │           └─ HashLookup(child: (SL3S5.BDNYB, SL3S5.M22QN), lookup: (sn.id, mf.M22QN))\n" +
-			" │               └─ CachedResults\n" +
```

Comment:
this one is cacheable. this change is explained by 1) i deleted the rule that uniformly applied hash lookups to every subquery, and 2) the join coster does not pick a hash join here only because every table is empty, every plan cost is zero, and we fall back to the original plan. I am fairly confident in practice we always choose hash over inner join. My follow-on work should resolve this confusion.